docs/scripts/buildApi.ts

```diff
@@ -192,6 +192,71 @@ async function annotateComponentDefinition(context: {
   writeFileSync(typesFilename, typesSourceNew, { encoding: 'utf8' });
 }
 
+function trimComment(comment: string) {
+  let i = 0;
+  for (; i < comment.length; i++) {
+    if (comment[i] !== '*' && comment[i] !== ' ') {
+      break;
+    }
+  }
+  return comment.substr(i, comment.length - 1);
+}
+
+async function updateStylesDefinition(context: { api: ReactApi; component: { filename: string } }) {
+  const { api, component } = context;
+
+  const typesFilename = component.filename.replace(/\.js$/, '.d.ts');
+  const typesSource = readFileSync(typesFilename, { encoding: 'utf8' });
+  const typesAST = await babel.parseAsync(typesSource, {
+    configFile: false,
+    filename: typesFilename,
+    presets: [require.resolve('@babel/preset-typescript')],
+  });
+
+  if (typesAST === null) {
+    throw new Error('No AST returned from babel.');
+  }
+
+  if (api.styles.classes.length === 0) {
+    const componentName = path.basename(typesFilename).replace(/\.d\.ts$/, '');
+
+    (typesAST as any).program.body.forEach((node: any) => {
+      const name = node.type === 'ExportNamedDeclaration' ? node?.declaration?.id?.name : undefined;
+      if (name === `${componentName}ClassKey` && node.declaration.typeAnnotation.types) {
+        const classes = node.declaration.typeAnnotation.types.map(
+          (node: babel.types.TSLiteralType) => node.literal.value,
+        );
+
+        const nodeLeadingComments = node.declaration.typeAnnotation.leadingComments || [];
+
+        node.declaration.typeAnnotation.types.forEach(
+          (typeNode: babel.types.TSLiteralType, idx: number) => {
+            let leadingComments = typeNode.leadingComments;
+            if (leadingComments) {
+              leadingComments = leadingComments.concat(nodeLeadingComments);
+            } else {
+              leadingComments = nodeLeadingComments;
+            }
+
+            if (leadingComments) {
+              for (let i = 0; i < leadingComments.length; i++) {
+                if (leadingComments[i].end + 5 === typeNode.literal.start) {
```

Comment:
Not sure if this is the best way to check if the comment corresponds to the specific node, the 5 comes from  the sequence between the comment end character and and the class key:
```
  /** Class name applied to the root element. */
  | 'root'
```
`/n  | `